integration-tests/narayana-jta/src/main/java/io/quarkus/narayana/jta/TransactionalResource.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.narayana.jta;
+
+import javax.transaction.Transaction;
+import javax.transaction.Transactional;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.arjuna.ats.arjuna.common.Uid;
+import com.arjuna.ats.jta.TransactionManager;
+
+@Path("/txuid")
+public class TransactionalResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    @Transactional
+    public String txuid() throws javax.transaction.SystemException {
+        Transaction txn = TransactionManager.transactionManager().getTransaction();
+        Uid uid = ((com.arjuna.ats.jta.transaction.Transaction) txn).get_uid();
+        return uid.toString();
```

Comment:
This test only says that there is a transaction. A more comprehensive test is to check that there is both a transaction and that it is active/running. So if you return `String.valueOf(txn.getStatus());` then that will test much more than the presence of a transaction. Then `TransactionalTestCase` can verify that it was running by checking that the response body has the value  `javax.transaction.Status.STATUS_ACTIVE (ie `given().when().get("/hello").then().body(is("0"));`). Here the value zero corresponds to STATUS_ACTIVE. To make the test clearer you could use the enum constant instead of "0".

Reply:
Thanks @mmusgrov - good pointing here !
I will add a new test case to check the transaction status.

Reply:
And if there is no transaction then the spec says that txn.getStatus() will throw SystemException and the test will fail.
Otherwise the PR looks good to go.

Reply:
> if there is no transaction associated with the calling thread then the spec says that txn.getStatus() will throw SystemException

@mmusgrov I can not find this describtion in the JTA spec. Also from the [javaee docs](https://docs.oracle.com/javaee/7/api/javax/transaction/Transaction.html#getStatus--)
```
int getStatus()
       throws SystemException
Obtain the status of the transaction associated with the target Transaction object.
Returns:
The transaction status. If no transaction is associated with the target object, this method returns the Status.NoTransaction value.
Throws:
SystemException - Thrown if the transaction manager encounters an unexpected error condition.
```
Do I miss something ?

Reply:
Ah yes, it will return Status.NoTransaction so the test will fail which is what we want.
Similarly if the call to getStatus throws a SystemException.

Reply:
But basically if the call to `getStatus` does not return `STATUS_ACTIVE` then the test will fail because the later test which checks the response will not match.

Reply:
sure, it makes sense.